### PR TITLE
LMN-4534 | Workouts missing data

### PR DESCRIPTION
**Pull request by NimGit**
Workouts missing data

https://metaflow.atlassian.net/browse/LMN-4534

### CODE REVIEW

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt [hunk 1]
```diff
@@ -6,92 +6,100 @@ import com.google.android.gms.auth.api.signin.GoogleSignIn
...
+        val gsa = GoogleSignIn.getAccountForExtension(currentActivity, authorizedFitnessOptions)
+
+        val request = DataReadRequest.Builder()
+            .read(workoutDataType)
             .enableServerQueries()
-            .readSessionsFromAllApps()
+            .aggregate(caloriesDataType)
+            .aggregate(activityDataType)
+            .setTimeRange(startTime, endTime, TimeUnit.MILLISECONDS)
+            .bucketByActivitySegment(29, TimeUnit.MINUTES) // It's not a workout if it's less than 29 minutes
```
Comment (rtokun) [inline android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt:47]: Just wondering where this number was taken from?

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt [hunk 2]
```diff
@@ -6,92 +6,100 @@ import com.google.android.gms.auth.api.signin.GoogleSignIn
...
             .addOnSuccessListener { response ->
-
-                val sessions = response.sessions.filter {
-                    it.getActivityTypeByReflection() !in listOf(
-                        unknownActivityType,
-                        stillActivityType,
-                        carActivityType
-                    )
+                val caloriesExpendedField = caloriesDataType.fields[0]
+                val activityField = activitySummaryDataType.fields.first {
```
Comment (rtokun) [inline android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt:55]: Please check if `first` doesn't throw an exception.
Maybe you have something like firstOrNull?

**Review by rtokun — APPROVED** [review]

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt (hunk 1 above)
Comment (NimGit) [reply android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt]: changed to 20

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt (hunk 2 above)
Comment (NimGit) [reply android/src/main/kotlin/com/metaflow/flutterhealthfit/WorkoutsReader.kt]: fixed

**Pull request merged**